addons/website_sale/models/product.py

```diff
@@ -141,11 +141,11 @@ class ProductTemplate(models.Model):
     website_public_price = fields.Float('Website public price', compute='_website_price', digits=dp.get_precision('Product Price'))
 
     def _website_price(self):
-        self.mapped('product_variant_id')
+        products = self.env['product.product'].browse(self.mapped('product_variant_id').ids)
```

Comment:
First filter out the ones that have no variant:

    self = self.filtered('product_variant_id')

Reply:
@rco-odoo A product template should have at least one variant: https://github.com/odoo/odoo/blob/10.0/addons/product/models/product_template.py#L112
So I guess we could simply get rid of this line.